---
title: "Breast Cancer"
output:
  html_vignette:
    mathjax: null
vignette: >
  %\VignetteIndexEntry{Breast Cancer}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
bibliography: cpdat.bib
csl: peerj.csl
---

```{r setup, include=FALSE}
library(canprot)
library(CHNOSZ)
library(knitr)
## use pngquant to reduce size of PNG images
knit_hooks$set(pngquant = hook_pngquant)
pngquant <- "--speed=1 --quality=0-25"
# in case pngquant isn't available (R-Forge?)
if (!nzchar(Sys.which("pngquant"))) pngquant <- NULL 
```

This vignette shows compositional metrics and phylostrata for proteins that are differentially expressed in breast cancer compared to normal tissue.
Abbreviations:

  * <i>Z</i><sub>C</sub> &ndash; carbon oxidation state; <i>n</i><sub>H<sub>2</sub>O</sub> &ndash; stoichiometric hydration state; <i>n</i><sub>O<sub>2</sub></sub> &ndash; stoichiometric oxidation state.
  * <i>n</i><sub>AA</sub> &ndash; protein length; PS &ndash; phylostrata.
  * <i>n</i><sub>down</sub> &ndash; number of down-regulated proteins; <i>n</i><sub>up</sub> &ndash; number of up-regulated proteins.

Stoichiometric values are derived from basis species using the rQEC method (Dick et al., 2020) or from amino acid biosynthetic reactions.
References for gene ages: [Trigos et al. (2017)](https://doi.org/10.1073/pnas.1617743114) (TPPG17); [Liebeskind et al. (2016)](https://doi.org/10.1093/gbe/evw113) (LMM16).

```{r options, echo=FALSE}
options(width = 90)
```

<style type="text/css">
body {
  max-width: 800px;
  margin-left: auto;
  margin-right: auto;
}
</style>

```{r datasets}
datasets <- pdat_breast(2020)
```

```{r comptab, results="hide", message=FALSE, echo = FALSE}
pdat1 <- lapply_canprot(datasets, pdat_breast)
comptab1 <- lapply(pdat1, get_comptab)
pdat2 <- lapply(pdat1, pdat_recomp, "biosynth")
comptab2 <- lapply(pdat2, get_comptab, "nO2")
comptab3 <- lapply(pdat1, get_comptab, "nAA", "PS")
comptab4 <- lapply(pdat1, get_comptab, "nAA", "PS", PS_source = "LMM16")
```

Dashed contour lines in the plots outline the 50% credible region for highest probability density.
Triangle symbols indicate transcriptomic datasets; all others are protein expression.

```{r diffplot, fig.width=8, fig.height=8, fig.align = "center", echo = FALSE, pngquant = pngquant}
par(mfrow = c(2, 2), mar = c(4, 4, 2, 2), mgp = c(2.5, 1, 0))
pch <- ifelse(grepl("transcript", datasets), 2, 1)
diffplot(comptab1, pch = pch)
title(quote("rQEC"~italic(n)[H[2]*O]), font.main = 1)
diffplot(comptab2, c("nO2", "nH2O"), pch = pch)
title(quote("Biosynthetic"~italic(n)[H[2]*O]~"and"~italic(n)[O[2]]), font.main = 1)
diffplot(comptab3, c("nAA", "PS"), pch = pch)
title("Trigos et al. (2017) ages", font.main = 1)
diffplot(comptab4, c("nAA", "PS"), pch = pch)
title("Liebeskind et al. (2016) ages", font.main = 1)
```

In the table, values of &Delta;<i>Z</i><sub>C</sub>, &Delta;<i>n</i><sub>H<sub>2</sub>O</sub> and &Delta;<i>n</i><sub>O<sub>2</sub></sub> are multiplied by 1000, values of &Delta;PS are multiplied by 100, and negative values are shown in bold.

```{r xsummary, results="asis", echo = FALSE}
library(xtable)
out <- xsummary2(comptab1, comptab2, comptab3, comptab4)
# round values and include dataset tags
tags <- sapply(sapply(strsplit(datasets, "="), "[", -1), paste, collapse = ";")
out <- cbind(out[, 1:2], tags = tags, out[, 3:11])
out[, 6:12] <- round(out[, 6:12], 4)
write.csv(out, "breast.csv", row.names = FALSE, quote = 2)
```

## Data Sources
Gene names or other identifiers were converted to UniProt accession numbers using the <a href="https://www.uniprot.org/mapping/">UniProt mapping tool</a>.

<b>a</b>. Extracted from Supporting Information RTF files of @AMG+08. Proteins identified by any number of peptides in both cancer and matched periphery were excluded; of the remaining proteins those identified by at least two peptides were used.
<b>b</b>. Table S2(a) of @CIR+10 (proteins differentially abundant at or above 99% confidence level).
<b>c</b>. Table 4 of @SRG+10.
<b>d</b>. Tables 1 and 2 of @HTP+11.
<b>e</b>. <b>f</b>. Supporting Table 6 of @GTM+12.
<b>g</b>. Table S1 of @LLL+13.
<b>h</b>. <b>i</b>. Table 2 of @SRS+13. For DCIS (3 patients), proteins were classified as up/down expressed if 2 or more ratios were greater/less than 1, and no ratios were less/greater than 1. For IC (4 patients), proteins were classified as up/down expressed if 3 or more ratios were greater/less than 1, and no ratios were less/greater than 1.
<b>j</b>. Extracted from Table S2 of @GSB+14. Values in all LFQ columns (distant, near, tumor) were quantile normalized (R function preprocessCore::normalize.quantiles) ([Bolstad et al., 2003](https://doi.org/10.1093/bioinformatics/19.2.185)), then the ratio between tumor and distant was calculated. Proteins with normalized LFQ ratios > 1.2 or < 1/1.2 and _p_-value < 0.05 were identified as differentially expressed.
<b>k</b>. Supplementary Tables 12, 13, 14 and 15 of @PPH+14, filtered to include proteins that are up- or down-regulated in all subtypes (LUM, LUMHER, HER, TN).
<b>l</b>. Supplemental Data S3 of @CVJ+15, filtered to include proteins with min and max credible intervals for expression ratios that are both <1 or >1.
<b>m</b>. <b>n</b>. <b>o</b>. Differential gene expression was calculated for [GSE57297](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE57297) [@FKZ+15] using the NCBI GEO2R service, with cutoffs for Benjamini and Hochberg adjusted _p_-value < 0.05 and log2 fold-change â‰¥ 2.
<b>p</b>. <b>q</b>. Supplementary Table S3 of @PGT+16.
<b>r</b>. Table S4A of @PBR+16.
<b>s</b>. Supplemental Table 4 of @BST+17, filtered to include proteins with _p_-value < 2.
<b>t</b>. <b>u</b>. Tables S5-1 (differentially expressed proteins for 52 tumor / non-cancerous tissue pairs) and S5-2 (13 basal-like tumor / non-cancerous tissue pairs) of @TZD+18, filtered to include proteins with log<sub>2</sub> fold change > 1 or < -1.
<b>v</b>. <b>w</b>. Supplementary File S1 of @GCS+19 (PT: primary breast tissue; NCT: non tumor contralateral breast tissue; ANT: non tumor adjacent breast tissue).
<b>x</b>. Supplementary Table 2 of @LLC+19.
<b>y</b>. Gene names from Supplementary Table S4 of @MBP+19.

## References
