---
title: "canprot: Chemical metrics of proteins"
output:
  html_vignette:
    mathjax: null
vignette: >
  %\VignetteIndexEntry{canprot: Chemical metrics of proteins}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
bibliography: canprot.bib
link-citations: yes
csl: elementa.csl
---

```{r setup, include=FALSE}
library(canprot)
library(CHNOSZ)
oldopt <- options(width = 72)
```

```{r HTML, include=FALSE}
Zc <- "<i>Z</i><sub>C</sub>"
nC <- "<i>n</i><sub>C</sub>"
nH2O <- "<i>n</i><sub>H<sub>2</sub>O</sub>"
H2O <- "H<sub>2</sub>O"
O2 <- "O<sub>2</sub>"
```

The **canprot** package has functions to calculate chemical metrics of proteins from their amino acid composition.
This vignette was compiled on `r Sys.Date()` with **canprot** version `r packageDescription("canprot")$Version`.

## Reading FASTA files

`KHAB17.fast` was obtained from Supplemental Information of @KHAB17 and is provided in the `extdata` directory of **canprot**.
Use `read.fasta()` to read the file and return a data frame of amino acid composition.

```{r read.fasta_KHAB17}
my_fasta_file <- system.file("extdata/fasta/KHAB17.fasta", package = "canprot")
aa <- read.fasta(my_fasta_file)
```

*To read your own FASTA file, just replace `my_fasta_file` with the path to your file.*

The result is small enough that we can look at it here.
The data frame has four columns for identifying information (`protein`, `organism`, `ref`, and `abbrv`); the first two are filled by `read.fasta()`.
The `chains` column is the number of polypeptide chains.

*None of the first five columns is necessary for the calculation of chemical metrics.*
They are provided for compatibility with CHNOSZ, and if you don't plan to use CHNOSZ, you put any information here that you want, including NA values, or remove the columns completely.
The columns that do matter for the calculation of chemical metrics are the last 20 columns, named by the 3-letter abbreviations of the amino acids.

These particular sequences are ancestral sequences of Rubisco.
A geochemical biology hypothesis is that the proteins should get more oxidized after the GOE.
Is this what happens?
Plot the carbon oxidation state (`r Zc`) to find out.

```{r Zc_KHAB17, out.width = "75%", fig.align = "center", fig.width = 6, fig.height = 4}
xlab <- "Ancestral sequences (older to younger)"
plot(Zc(aa), type = "b", xaxt = "n", xlab = xlab, ylab = cplab$Zc)
names <- gsub(".*_", "", aa$protein)
axis(1, at = 1:6, names)
abline(v = 3.5, lty = 2, col = 8)
axis(3, at = 3.5, "GOE")
```

Note the use of pre-formatted plot labels for chemical metrics available in `cplab`.
The vertical line denotes the proposed timing of the Great Oxidation Event (GOE) between Anc. I/III and Anc. I [@KHAB17].
**This analysis shows that proteins do get more oxidized after the GOE.**

## Human proteins in **canprot**

**canprot** has a database of amino acid compositions of human proteins assembled from UniProt.
Use `protcomp()` to get the amino acid composition.
This example is for alanine amino-transferase, which has a UniProt ID of `P24298`:
```{r protcomp_ALAT1}
(pc <- protcomp("P24298"))
Zc(pc$aa)
```

Do you have a list of UniProt IDs for a differential expression dataset?
Great!
We can use those to calculate chemical metrics and make a boxplot.
The IDs in this example come from Figure 5 of @DKM+20, where they were identified as down- and up-expressed proteins in aggregate (or 3D) cell culture compared to monolayer (or 2D) culture.

```{r DKM20_UniProt}
up <-   c("Q92743", "P43490", "P52895", "P98160", "P23142", "P17301",
"U3KQK0", "Q15582", "Q9HCJ1", "P36222", "P27701", "Q08380", "P08572",
"P00734", "P22413", "O43657", "P35625", "O75348", "P02649", "P13861",
"P10620", "Q9H3N1", "A8K878", "P13611", "P07305", "E7ESP4", "Q9Y625",
"Q5ZPR3", "P62266", "Q96AQ6", "Q8N357", "Q13217", "Q9Y230", "Q9Y639",
"Q86W92", "C9JF17", "Q96PK6", "O95671", "P01033", "Q13501", "P69905",
"Q9Y5X1", "P50281", "Q9UBG0", "O60831", "P02751", "O43854", "P61803",
"J3KN66", "P42765", "P36543", "P15121", "Q16563", "Q12884", "P27695",
"P12110", "P07686", "Q92598", "Q02818", "Q07954", "O60493", "P40939",
"Q9Y3I0", "P51149", "P46776", "P46778", "P62805")

down <- c("J3KN67", "Q9Y490", "J3KNQ4", "E7EVA0", "Q01082", "J3KQ32",
"P54136", "Q9Y696", "Q01995", "Q15404", "P62714", "Q09666", "P07814",
"E7EQR4", "P46821", "O75369", "P02452", "P08123", "P54577", "P01023",
"Q6ZN40", "P42224", "B4DUT8", "Q13443", "Q9HCE1", "Q6DKJ4", "P50552",
"P35222", "P20908", "Q15417", "O75822", "P17812", "P05997", "P04080",
"O43294", "P08243", "P02458")
```

With those UniProt IDs for human proteins we can retrieve the amino acid compositions, then calculate a couple of chemical metrics and make some boxplots comparing the groups of down- and up-expressed proteins.

```{r DKM20_plot, out.width = "75%", fig.align = "center", fig.width = 7, fig.height = 5}
aa_down <- protcomp(down)$aa
aa_up <- protcomp(up)$aa
bp_names <- paste0(c("Down (", "Up ("), c(nrow(aa_down), nrow(aa_up)), c(")", ")"))

par(mfrow = c(1, 2))

Zclist <- list(Zc(aa_down), Zc(aa_up))
names(Zclist) <- bp_names
boxplot(Zclist, ylab = cplab$Zc, col = c(3, 2))
names(Zclist) <- c("x", "y")
p <- do.call(wilcox.test, Zclist)$p.value
legend("top", paste("p =", round(p, 3)), bty = "n")
title("Cabon oxidation state", font.main = 1)

nH2Olist <- list(nH2O(aa_down), nH2O(aa_up))
names(nH2Olist) <- bp_names
boxplot(nH2Olist, ylab = cplab$nH2O, col = c(3, 2))
names(nH2Olist) <- c("x", "y")
p <- do.call(wilcox.test, nH2Olist)$p.value
legend("top", paste("p =", round(p, 3)), bty = "n")
title("Stoichiometric hydration state", font.main = 1)
```

The colors are chosen because green and red traditionally are used for down- and up-expressed proteins.
According to this, there is no significant difference of `r Zc` between the groups of down- and up-expressed proteins.
In contrast, `r nH2O` is significantly lower for up-expressed compared to down-expresssed proteins.

**This dehydration trend characterizes most datasets for 3D cell culture [@Dic21a].**
*The differential expression datasets analyzed in that paper, which were previously in **canprot**, have been moved to [JMDplots](https://github.com/jedick/JMDplots).*

## More details on chemical metrics

Use the functions `Zc` and `nH2O` to calculate the carbon oxidation state (`r Zc`) and stoichiometric hydration state (`r nH2O`) of the molecules.
With default settings, `r nH2O` is calculated from a theoretical overall reaction to form the protein from the basis species glutamine, glutamic acid, cysteine, `r H2O`, and `r O2`, abbreviated as **QEC** (the setting can be viewed with `getOption("basis")`).
To see how this works, consider the formation reaction of alanylglycine, which can be written using functions in the **CHNOSZ** package:
```{r AG_reaction, message = FALSE}
CHNOSZ::basis("QEC")
CHNOSZ::subcrt("alanylglycine", 1)$reaction
```

Alanylglycine has the same formula as glutamine, so there is no water in the reaction, and `r nH2O` is zero.
Let's do the calculation with the `nH2O()` function to see this result.
We have to specify `terminal_H2O = 1` in order to account for the terminal -H and -OH groups.
The default is 0 because, most of the time, we are interested in the per-amino-acid contributions for proteins.
```{r AG_metrics}
AG <- data.frame(Ala = 1, Gly = 1)
Zc(AG)
nH2O(AG, terminal_H2O = 1)
```

Note that the value of `r nH2O` is normalized by protein length (i.e. it is divided by the number of amino acid residues), but carbon oxidation state is a per-carbon value, so it is not divided by the number of residues.

Now let's try an actual protein, chicken egg-white lysozome, which has the name LYSC\_CHICK in UniProt with accession number [P00698](https://www.uniprot.org/uniprot/P00698).
The amino acid compositions of this and selected other proteins are available in the **CHNOSZ** package.
Here we get the amino acid composition and print the protein length and `r Zc` and `r nH2O`:
```{r LYSC_CHICK_metrics}
AA <- CHNOSZ::pinfo(CHNOSZ::pinfo("LYSC_CHICK"))
plength(AA)
Zc(AA)
nH2O(AA)
```

To see where the value of `r nH2O` comes from, write the formation reaction of LYSC\_CHICK from the **QEC** basis species.
```{r LYSC_reaction, message = FALSE}
CHNOSZ::subcrt("LYSC_CHICK", 1)$reaction
```

This shows that `r with(CHNOSZ::subcrt("LYSC_CHICK", 1)$reaction, coeff[name == "water"])` water molecules are released in the reaction.
`r nH2O` is the opposite of this value (because we are counting how many `r H2O` units go **into** forming the protein), divided by the length of the protein (`r plength(AA)`), which gives us the value of `r nH2O`: `r nH2O(AA)`.

`nH2O` works not by writing the formation reaction for each protein but rather by using precomputed values of `r nH2O` for each amino acid.
The two methods give equivalent results, as described in @DYT20.

Similarly, `Zc` uses precomputed values of `r Zc` and `r nC` (number of carbon atoms) for each amino acid.
**NOTE**: Calculating `r Zc` of proteins from amino acid frequencies (i.e. abundances or counts in a protein sequence) requires weighting the amino-acid `r Zc` by the number of carbon atoms in each amino acid, in addition to weighting by amino acid frequency.
Using the unweighted mean of `r Zc` of amino acids is a common mistake that leads to artificially higher values for proteins.

### Other chemical metrics

There are also functions for calculating the grand average of hydropathicity (GRAVY, which is higher for proteins with more hydrophobic amino acids) and isoelectric point (pI) of proteins.
There are some limitations of this implementation [see @DYT20 for details], but values for representative proteins are equal to those computed with the ProtParam tool [@GHG+05] in UniProt (see `?pI` for numerical tests).
```{r GRAVY_pI}
proteins <- c("LYSC_CHICK", "RNAS1_BOVIN", "AMYA_PYRFU")
AA <- CHNOSZ::pinfo(CHNOSZ::pinfo(proteins))
pI(AA)
GRAVY(AA)
```

```{r reset, include=FALSE}
options(oldopt)
```

## References
