---
title: "Colorectal Cancer"
output:
  html_vignette:
    mathjax: null
vignette: >
  %\VignetteIndexEntry{Colorectal Cancer}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
bibliography: cpdat.bib
csl: peerj.csl
---

```{r setup, include=FALSE}
library(canprot)
library(CHNOSZ)
library(knitr)
## use pngquant to reduce size of PNG images
knit_hooks$set(pngquant = hook_pngquant)
pngquant <- "--speed=1 --quality=0-25"
# in case pngquant isn't available (R-Forge?)
if (!nzchar(Sys.which("pngquant"))) pngquant <- NULL 
```

This vignette shows compositional metrics and phylostrata for proteins that are differentially expressed in colorectal cancer compared to normal tissue.
Abbreviations:

  * <i>Z</i><sub>C</sub> &ndash; carbon oxidation state; <i>n</i><sub>H<sub>2</sub>O</sub> &ndash; stoichiometric hydration state; <i>n</i><sub>O<sub>2</sub></sub> &ndash; stoichiometric oxidation state.
  * <i>n</i><sub>AA</sub> &ndash; protein length; PS &ndash; phylostrata.
  * <i>n</i><sub>down</sub> &ndash; number of down-regulated proteins; <i>n</i><sub>up</sub> &ndash; number of up-regulated proteins.

Stoichiometric values are calculated using basis species (rQEC method; Dick et al., 2020) or amino acid biosynthetic reactions.
References for gene ages: [Trigos et al. (2017)](https://doi.org/10.1073/pnas.1617743114) (TPPG17); [Liebeskind et al. (2016)](https://doi.org/10.1093/gbe/evw113) (LMM16).

```{r options, echo=FALSE}
options(width = 90)
```

<style type="text/css">
body {
  max-width: 800px;
  margin-left: auto;
  margin-right: auto;
}
</style>

```{r datasets}
datasets <- pdat_colorectal(2020)
```

```{r comptab, results="hide", message=FALSE, echo = FALSE}
pdat1 <- lapply_canprot(datasets, pdat_colorectal)
comptab1 <- lapply(pdat1, get_comptab)
pdat2 <- lapply(pdat1, pdat_recomp, "biosynth")
comptab2 <- lapply(pdat2, get_comptab, "nO2")
comptab3 <- lapply(pdat1, get_comptab, "nAA", "PS")
comptab4 <- lapply(pdat1, get_comptab, "nAA", "PS", PS_source = "LMM16")
```

Dashed contour lines in the plots outline the 50% credible region for highest probability density.
Triangle symbols indicate transcriptomic datasets; all others are protein expression.

```{r diffplot, fig.width=8, fig.height=8, fig.align = "center", echo = FALSE, pngquant = pngquant}
par(mfrow = c(2, 2), mar = c(4, 4, 2, 2), mgp = c(2.5, 1, 0))
pch <- ifelse(grepl("transcript", datasets), 2, 1)
diffplot(comptab1, pch = pch)
title(quote("rQEC"~italic(n)[H[2]*O]), font.main = 1)
diffplot(comptab2, c("nO2", "nH2O"), pch = pch)
title(quote("Biosynthetic"~italic(n)[H[2]*O]~"and"~italic(n)[O[2]]), font.main = 1)
diffplot(comptab3, c("nAA", "PS"), pch = pch)
title("Trigos et al. (2017) ages", font.main = 1)
diffplot(comptab4, c("nAA", "PS"), pch = pch)
title("Liebeskind et al. (2016) ages", font.main = 1)
```

In the table, values of &Delta;<i>Z</i><sub>C</sub>, &Delta;<i>n</i><sub>H<sub>2</sub>O</sub> and &Delta;<i>n</i><sub>O<sub>2</sub></sub> are multiplied by 1000, values of &Delta;PS are multiplied by 100, and negative values are shown in bold.

```{r xsummary, results="asis", echo = FALSE}
library(xtable)
out <- xsummary2(comptab1, comptab2, comptab3, comptab4)
# round values and include dataset tags
tags <- sapply(sapply(strsplit(datasets, "="), "[", -1), paste, collapse = ";")
out <- cbind(out[, 1:2], tags = tags, out[, 3:11])
out[, 6:12] <- round(out[, 6:12], 4)
write.csv(out, "colorectal.csv", row.names = FALSE, quote = 2)
```

## Data Sources
Gene names or other identifiers were converted to UniProt accession numbers using the <a href="https://www.uniprot.org/mapping/">UniProt mapping tool</a>, except for IPI accession numbers, which were converted using the <a href="https://david-d.ncifcrf.gov/conversion.jsp">DAVID 6.7 conversion tool</a>.

__a__. Genes identified as up- or down-expressed in human CRC in Table 4 of @KPF+07.
__b__. Table 1 and Supplementary Data 1 of @WTK+08 (Swiss-Prot and UniProt accession numbers from Supplementary Data 2).
__c__. __d__. IPI accession numbers from Supplemental Table 4 of @XZC+10.
__e__. IPI accession numbers from Supplemental Table 4 of @ZYS+10.
__f__. __g__. __h__. __i__. Gene names from supplemental Table 9 of @BPV+11.
__j__. Supplementary Table 2 of @JCF+11.
__k__. Table S8 of @MRK+11.
__l__. Supplementary Table 1 of @SHHS11.
__m__. Gene names from Supplementary Table 4 of @KYK+12, filtered to include proteins with expression ratio > 2 or < 0.5 in both mTRAQ and cICAT analyses.
__n__. Supplementary Table 4 of @WOD+12.
__o__. Table 2 of @CZD+14.
__p__. Ensembl protein IDs from Supporting Table 2 of @STK+15.
__q__. Proteins marked as having a significant change between normal tissue (N) and adenocarcinoma (C) in SI Table 3 of @WDO+15.
__r__. SI Table S3 of @LXM+16, filtered to include proteins with _p_-value < 0.05.
__s__. __t__. Gene names from Supplementary Table 4 of @PHL+16, for differential expression between normal colonic mucosa (NC) and carcinoma in situ (CIS) or invasive colorectal cancer (ICC).
__u__. Table S3 of @CTW+17.
__v__. Dataset 6A of @HZW+17.
__w__. Supplementary Material Table S1 of @LLL+17.
__x__. Table 2 of @NKG+17.
__y__. Table S2 of @QMB+17, filtered to include comparisons between adenocarcinoma and diverticular disease.
__z__. Table S1 of @TMS+17, filtered to include proteins that are consistently up- or down-expressed in at least 11 of 12 patients.
__A__. IPI accession numbers from Table S2 of @ZLY+17.
__B__. Supplementary Table 1B of @AKG+18, filtered to include proteins with expression ratio > 3/2 or < 2/3.
__C__. __D__. Supplementary Table S2 of @STA+19, filtered to include proteins with log~2~ ratio > Â± the standard deviation of values for all quantified proteins.
__E__. Supplementary Information Table S2 of @WYL+19.

## References
