<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">



<title>Data Sources</title>






<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">Data Sources</h1>



<p>This is a description of the sources of datasets for protein expression in colorectal cancer contained in this package. The text is taken from <a href="http://dx.doi.org/10.7717/peerj.2238">Dick, 2016</a>.</p>
<hr />
<p><code>WTK+08</code>: <a href="http://dx.doi.org/10.1002/prca.200780111">Watanabe et al. (2008)</a> used 2-nitrobenzenesulfenyl labeling and MS/MS analysis to identify 128 proteins with differential expression in paired CRC and normal tissue specimens from 12 patients. The list of proteins used in this study was generated by combining the lists of up- and down-regulated proteins from Table 1 and Supplementary Data 1 of <span class="citation">Watanabe et al. (2008)</span> with the Swiss-Prot and UniProt accession numbers from their Supplementary Data 2.</p>
<p><code>AKP+10</code>: <a href="http://dx.doi.org/10.1074/mcp.M900546-MCP200">Albrethsen et al. (2010)</a> used nano-LC-MS/MS to characterize proteins from the nuclear matrix fraction in samples from 2 patients each with adenoma (ADE), chromosomal instability CRC (CIN+) and microsatellite instability CRC (MIN+). Cluster analysis was used to classify proteins with differential expression between ADE and CIN+, MIN+, or in both subtypes of carcinoma (CRC). Here, gene names from Supplementary Tables 5–7 of <span class="citation">Albrethsen et al. (2010)</span> were converted to UniProt IDs using the UniProt mapping tool.</p>
<p><code>JKMF10</code>: <a href="http://dx.doi.org/10.1016/j.jprot.2010.06.004">Jimenez et al. (2010)</a> compiled a list of candidate serum biomarkers from a meta-analysis of the literature. In the meta-analysis, 99 up- or down-expressed proteins were identified in at least 2 studies. The list of UniProt IDs used in this study was taken from Table 4 of <span class="citation">Jimenez et al. (2010)</span>.</p>
<p><code>XZC+10</code>: <a href="http://dx.doi.org/10.1021/pr100406z">Xie et al. (2010)</a> used a gel-enhanced LC-MS method to analyze proteins in pooled tissue samples from 13 stage I and 24 stage II CRC patients and pooled normal colonic tissues from the same patients. Here, IPI accession numbers from Supplemental Table 4 of <span class="citation">Xie et al. (2010)</span> were converted to UniProt IDs using the DAVID conversion tool.</p>
<p><code>ZYS+10</code>: <a href="http://dx.doi.org/10.1111/j.1349-7006.2009.01424.x">Zhang et al. (2010)</a> used acetylation stable isotope labeling and LTQ-FT MS to analyze proteins in pooled microdissected epithelial samples of tumor and normal mucosa from 20 patients, finding 67 and 70 proteins with increased or decreased expression (ratios ≥2 or ≤0.5). Here, IPI accession numbers from Supplemental Table 4 of <span class="citation">Zhang et al. (2010)</span> were converted to UniProt IDs using the DAVID conversion tool.</p>
<p><code>BPV+11</code>: <a href="http://dx.doi.org/10.1074/mcp.M111.009712">Besson et al. (2011)</a> analyzed microdissected cancer and normal tissues from 28 patients (4 adenoma samples and 24 CRC samples at different stages) using iTRAQ labeling and MALDI-TOF/TOF MS to identify 555 proteins with differential expression between adenoma and stage I, II, III, IV CRC. Here, gene names from supplemental Table 9 of <span class="citation">Besson et al. (2011)</span> were converted to UniProt IDs using the UniProt mapping tool.</p>
<p><code>JCF+11</code>: <a href="http://dx.doi.org/10.1039/C1MB05236E">Jankova et al. (2011)</a> analyzed paired samples from 16 patients using iTRAQ-MS to identify 118 proteins with &gt;1.3-fold differential expression between CRC tumors and adjacent normal mucosa. The protein list used in this study was taken from Supplementary Table 2 of <span class="citation">Jankova et al. (2011)</span>.</p>
<p><code>MRK+11</code>: <a href="http://dx.doi.org/10.1007/s10142-010-0200-5">Mikula et al. (2011)</a> used iTRAQ labeling with LC-MS/MS to identify a total of 1061 proteins with differential expression (fold change ≥1.5 and false discovery rate ≤0.01) between pooled samples of 4 normal colon (NC), 12 tubular or tubulo-villous adenoma (AD) and 5 adenocarcinoma (AC) tissues. The list of proteins used in this study was taken from from Table S8 of <span class="citation">Mikula et al. (2011)</span>.</p>
<p><code>KKL+12</code>: <a href="http://dx.doi.org/10.1016/j.jprot.2011.12.002">Kim et al. (2012)</a> used difference in-gel electrophoresis (DIGE) and cleavable isotope-coded affinity tag (cICAT) labeling followed by mass spectrometry to identify 175 proteins with more than 2-fold abundance ratios between microdissected and pooled tumor tissues from stage-IV CRC patients with good outcomes (survived more than five years; 3 patients) and poor outcomes (died within 25 months; 3 patients). The protein list used in this study was made by filtering the cICAT data from Supplementary Table 5 of <span class="citation">Kim et al. (2012)</span> with an abundance ratio cutoff of &gt;2 or &lt;0.5, giving 147 proteins. IPI accession numbers were converted to UniProt IDs using the DAVID conversion tool.</p>
<p><code>KYK+12</code>: <a href="http://dx.doi.org/10.1016/j.jprot.2011.11.021">Kang et al. (2012)</a> used mTRAQ and cICAT analysis of pooled microsatellite stable (MSS-type) CRC tissues and pooled matched normal tissues from 3 patients to identify 1009 and 478 proteins in cancer tissue with increased or decreased expression by higher than 2-fold, respectively. Here, the list of proteins from Supplementary Table 4 of <span class="citation">Kang et al. (2012)</span> was filtered to include proteins with expression ratio &gt;2 or &lt;0.5 in both mTRAQ and cICAT analyses, leaving 175 up-expressed and 248 down-expressed proteins in CRC. Gene names were converted to UniProt IDs using the UniProt mapping tool.</p>
<p><code>WOD+12</code>: <a href="http://dx.doi.org/10.1038/msb.2012.44">Wiśniewski et al. (2012)</a> used LC-MS/MS to analyze proteins in microdissected samples of formalin-fixed paraffin-embedded (FFPE) tissue from 8 patients; at <em>P</em> &lt; 0.01, 762 proteins had differential expression between normal mucosa and primary tumors. The list of proteins used in this study was taken from Supplementary Table 4 of <span class="citation">Wiśniewski et al. (2012)</span>.</p>
<p><code>YLZ+12</code>: <a href="http://dx.doi.org/10.1021/pr300020p">Yao et al. (2012)</a> analyzed the conditioned media of paired stage I or IIA CRC and normal tissues from 9 patients using lectin affinity capture for glycoprotein (secreted protein) enrichment by nano LC-MS/MS to identify 68 up-regulated and 55 down-regulated differentially expressed proteins. IPI accession numbers listed in Supplementary Table 2 of <span class="citation">Yao et al. (2012)</span> were converted to UniProt IDs using the DAVID conversion tool.</p>
<p><code>MCZ+13</code>: <a href="http://dx.doi.org/10.1002/elps.201200596">Mu et al. (2013)</a> used laser capture microdissection (LCM) to separate stromal cells from 8 colon adenocarcinoma and 8 non-neoplastic tissue samples, which were pooled and analyzed by iTRAQ to identify 70 differentially expressed proteins. Here, gi numbers listed in Table 1 of <span class="citation">Mu et al. (2013)</span> were converted to UniProt IDs using the UniProt mapping tool; FASTA sequences of 31 proteins not found in UniProt were downloaded from NCBI and amino acid compositions were added to <a href="../html/human.html">human_extra.csv</a>.</p>
<p><code>KWA+14</code>: <a href="http://dx.doi.org/10.1016/j.bbapap.2013.12.006">Knol et al. (2014)</a> used differential biochemical extraction to isolate the chromatin-binding fraction in frozen samples of colon adenomas (3 patients) and carcinomas (5 patients), and LC-MS/MS was used for protein identification and label-free quantification. The results were combined with a database search to generate a list of 106 proteins with nuclear annotations and at least a three-fold expression difference. Here, gene names from Table 2 of <span class="citation">Knol et al. (2014)</span> were converted to UniProt IDs.</p>
<p><code>UNS+14</code>: <a href="http://dx.doi.org/10.1074/mcp.M113.035105">Uzozie et al. (2014)</a> analyzed 30 samples of colorectal adenomas and paired normal mucosa using iTRAQ labeling, OFFGEL electrophoresis and LC-MS/MS. 111 proteins with expression fold changes (log<sub>2</sub>) at least +/- 0.5 and statistical significance threshold <em>q</em> &lt; 0.02 that were also quantified in cell-line experiments were classified as “epithelial cell signature proteins”. UniProt IDs were taken from Table III of <span class="citation">Uzozie et al. (2014)</span>.</p>
<p><code>WKP+14</code>: <a href="http://dx.doi.org/10.1016/j.jprot.2014.01.001">de Wit et al. (2014)</a> analyzed the secretome of paired CRC and normal tissue from 4 patients, adopting a five-fold enrichment cutoff for identification of candidate biomarkers. Here, the list of proteins from Supplementary Table 1 of <span class="citation">Wit et al. (2014)</span> was filtered to include those with at least five-fold greater or lower abundance in CRC samples and <em>p</em> &lt; 0.05. Two proteins listed as “Unmapped by Ingenuity” were removed, and gene names were converted to UniProt IDs using the UniProt mapping tool.</p>
<p><code>STK+15</code>: <a href="http://dx.doi.org/10.1016/j.jprot.2015.05.037">Sethi et al. (2015)</a> analyzed the membrane-enriched proteome from tumor and adjacent normal tissues from 8 patients using label-free nano-LC-MS/MS to identify 184 proteins with a fold change &gt; 1.5 and <em>p</em>-value &lt; 0.05. Here, protein identifiers from Supporting Table 2 of <span class="citation">Sethi et al. (2015)</span> were used to find the corresponding UniProt IDs.</p>
<p><code>WDO+15</code>: <a href="http://dx.doi.org/10.1021/acs.jproteome.5b00523">Wiśniewski et al. (2015)</a> analyzed 8 matched formalin-fixed and paraffin-embedded (FFPE) samples of normal tissue (N) and adenocarcinoma (C) and 16 nonmatched adenoma samples (A) using LC-MS to identify 2300 (N/A), 1780 (A/C) and 2161 (N/C) up- or down-regulated proteins at <em>p</em> &lt; 0.05. The list of proteins used in this study includes only those marked as having a significant change in SI Table 3 of <span class="citation">Wiśniewski et al. (2015)</span>.</p>
<p><code>LPL+16</code>: <a href="http://dx.doi.org/10.18632/oncotarget.9362">Li et al. (2016)</a> used iTRAQ and 2D LC-MS/MS to analyze pooled samples of stroma purified by laser capture microdissection (LCM) from 5 cases of non-neoplastic colonic mucosa (NC), 8 of adenomatous colon polyps (AD), 5 of colon carcinoma in situ (CIS) and 9 of invasive colonic carcinoma (ICC). A total of 222 differentially expressed proteins between NC and other stages were identified. Here, gene symbols from Supplementary Table S3 of <span class="citation">Li et al. (2016)</span> were converted to UniProt IDs using the UniProt mapping tool.</p>
<p><code>PHL+16</code>: <a href="http://dx.doi.org/10.3748/wjg.v22.i18.4515">Peng et al. (2016)</a> used iTRAQ 2D LC-MS/MS to analyze pooled samples from 5 cases of normal colonic mucosa (NC), 8 of adenoma (AD), 5 of carcinoma in situ (CIS) and 9 of invasive colorectal cancer (ICC). A total of 326 proteins with differential expression between two successive stages (and, for CIS and ICC, also differentially expressed with respect to NC) were detected. The list of proteins used in this study was generated by converting the gene names in Supplementary Table 4 of <span class="citation">Peng et al. (2016)</span> to UniProt IDs using the UniProt mapping tool.</p>
<div id="references" class="section level2 unnumbered">
<h2>References</h2>
<div id="refs" class="references">
<div id="ref-AKP+10">
<p>Albrethsen J., Knol JC., Piersma SR., Pham TV., Wit M de., Mongera S., Carvalho B., Verheul HMW., Fijneman RJA., Meijer GA., Jimenez CR. 2010. Subnuclear proteomics in colorectal cancer: Identification of proteins enriched in the nuclear matrix fraction and regulation in adenoma to carcinoma progression. <em>Molecular &amp; Cellular Proteomics</em> 9:988–1005. DOI: <a href="https://doi.org/10.1074/mcp.M900546-MCP200">10.1074/mcp.M900546-MCP200</a>.</p>
</div>
<div id="ref-BPV+11">
<p>Besson D., Pavageau A-H., Valo I., Bourreau A., Bélanger A., Eymerit-Morin C., Moulière A., Chassevent A., Boisdron-Celle M., Morel A., Solassol J., Campone M., Gamelin E., Barré B., Coqueret O., Guette C. 2011. A quantitative proteomic approach of the different stages of colorectal cancer establishes OLFM4 as a new nonmetastatic tumor marker. <em>Molecular &amp; Cellular Proteomics</em> 10:M111.009712. DOI: <a href="https://doi.org/10.1074/mcp.M111.009712">10.1074/mcp.M111.009712</a>.</p>
</div>
<div id="ref-Dic16">
<p>Dick JM. 2016. Proteomic indicators of oxidation and hydration state in colorectal cancer. <em>PeerJ</em> 4:e2238. DOI: <a href="https://doi.org/10.7717/peerj.2238">10.7717/peerj.2238</a>.</p>
</div>
<div id="ref-JCF+11">
<p>Jankova L., Chan C., Fung CLS., Song X., Kwun SY., Cowley MJ., Kaplan W., Dent OF., Bokey EL., Chapuis PH., Baker MS., Robertson GR., Clarke SJ., Molloy MP. 2011. Proteomic comparison of colorectal tumours and non-neoplastic mucosa from paired patient samples using iTRAQ mass spectrometry. <em>Molecular Biosystems</em> 7:2997–3005. DOI: <a href="https://doi.org/10.1039/C1MB05236E">10.1039/C1MB05236E</a>.</p>
</div>
<div id="ref-JKMF10">
<p>Jimenez CR., Knol JC., Meijer GA., Fijneman RJA. 2010. Proteomics of colorectal cancer: Overview of discovery studies and identification of commonly identified cancer-associated proteins and candidate CRC serum markers. <em>Journal of Proteomics</em> 73:1873–1895. DOI: <a href="https://doi.org/10.1016/j.jprot.2010.06.004">10.1016/j.jprot.2010.06.004</a>.</p>
</div>
<div id="ref-KYK+12">
<p>Kang U-B., Yeom J., Kim H-J., Kim H., Lee C. 2012. Expression profiling of more than 3500 proteins of MSS-type colorectal cancer by stable isotope labeling and mass spectrometry. <em>Journal of Proteomics</em> 75:3050–3062. DOI: <a href="https://doi.org/10.1016/j.jprot.2011.11.021">10.1016/j.jprot.2011.11.021</a>.</p>
</div>
<div id="ref-KKL+12">
<p>Kim H-J., Kang U-B., Lee H., Jung J-H., Lee S-T., Yu M-H., Kim H., Lee C. 2012. Profiling of differentially expressed proteins in stage IV colorectal cancers with good and poor outcomes. <em>Journal of Proteomics</em> 75:2983–2997. DOI: <a href="https://doi.org/10.1016/j.jprot.2011.12.002">10.1016/j.jprot.2011.12.002</a>.</p>
</div>
<div id="ref-KWA+14">
<p>Knol JC., Wit M de., Albrethsen J., Piersma SR., Pham TV., Mongera S., Carvalho B., Fijneman RJA., Meijer GA., Jiménez CR. 2014. Proteomics of differential extraction fractions enriched for chromatin-binding proteins from colon adenoma and carcinoma tissues. <em>Biochimica et Biophysica Acta (BBA) - Proteins and Proteomics</em> 1844:1034–1043. DOI: <a href="https://doi.org/10.1016/j.bbapap.2013.12.006">10.1016/j.bbapap.2013.12.006</a>.</p>
</div>
<div id="ref-LPL+16">
<p>Li M., Peng F., Li G., Fu Y., Huang Y., Chen Z., Chen Y. 2016. Proteomic analysis of stromal proteins in different stages of colorectal cancer establishes Tenascin-C as a stromal biomarker for colorectal cancer metastasis. <em>Oncotarget</em> 7. DOI: <a href="https://doi.org/10.18632/oncotarget.9362">10.18632/oncotarget.9362</a>.</p>
</div>
<div id="ref-MRK+11">
<p>Mikula M., Rubel T., Karczmarski J., Goryca K., Dadlez M., Ostrowski J. 2011. Integrating proteomic and transcriptomic high-throughput surveys for search of new biomarkers of colon tumors. <em>Functional and Integrative Genomics</em> 11:215–224. DOI: <a href="https://doi.org/10.1007/s10142-010-0200-5">10.1007/s10142-010-0200-5</a>.</p>
</div>
<div id="ref-MCZ+13">
<p>Mu Y., Chen Y., Zhang G., Zhan X., Li Y., Liu T., Li G., Li M., Xiao Z., Gong X., Chen Z. 2013. Identification of stromal differentially expressed proteins in the colon carcinoma by quantitative proteomics. <em>Electrophoresis</em> 34:1679–1692. DOI: <a href="https://doi.org/10.1002/elps.201200596">10.1002/elps.201200596</a>.</p>
</div>
<div id="ref-PHL+16">
<p>Peng F., Huang Y., Li M-Y., Li G-Q., Huang H-C., Guan R., Chen Z-C., Liang S-P., Chen Y-H. 2016. Dissecting characteristics and dynamics of differentially expressed proteins during multistage carcinogenesis of human colorectal cancer. <em>World Journal of Gastroenterology</em> 22:4515–4528. DOI: <a href="https://doi.org/10.3748/wjg.v22.i18.4515">10.3748/wjg.v22.i18.4515</a>.</p>
</div>
<div id="ref-STK+15">
<p>Sethi MK., Thaysen-Andersen M., Kim H., Park CK., Baker MS., Packer NH., Paik Y-K., Hancock WS., Fanayan S. 2015. Quantitative proteomic analysis of paired colorectal cancer and non-tumorigenic tissues reveals signature proteins and perturbed pathways involved in CRC progression and metastasis. <em>Journal of Proteomics</em> 126:54–67. DOI: <a href="https://doi.org/10.1016/j.jprot.2015.05.037">10.1016/j.jprot.2015.05.037</a>.</p>
</div>
<div id="ref-UNS+14">
<p>Uzozie A., Nanni P., Staiano T., Grossmann J., Barkow-Oesterreicher S., Shay JW., Tiwari A., Buffoli F., Laczko E., Marra G. 2014. Sorbitol dehydrogenase overexpression and other aspects of dysregulated protein expression in human precancerous colorectal neoplasms: A quantitative proteomics study. <em>Molecular &amp; Cellular Proteomics</em> 13:1198–1218. DOI: <a href="https://doi.org/10.1074/mcp.M113.035105">10.1074/mcp.M113.035105</a>.</p>
</div>
<div id="ref-WTK+08">
<p>Watanabe M., Takemasa I., Kawaguchi N., Miyake M., Nishimura N., Matsubara T., Matsuo E-i., Sekimoto M., Nagai K., Matsuura N., Monden M., Nishimura O. 2008. An application of the 2-nitrobenzenesulfenyl method to proteomic profiling of human colorectal carcinoma: A novel approach for biomarker discovery. <em>Proteomics: Clinical Applications</em> 2:925–935. DOI: <a href="https://doi.org/10.1002/prca.200780111">10.1002/prca.200780111</a>.</p>
</div>
<div id="ref-WOD+12">
<p>Wiśniewski JR., Ostasiewicz P., Duś K., Zielińska DF., Gnad F., Mann M. 2012. Extensive quantitative remodeling of the proteome between normal colon tissue and adenocarcinoma. <em>Molecular Systems Biology</em> 8:611. DOI: <a href="https://doi.org/10.1038/msb.2012.44">10.1038/msb.2012.44</a>.</p>
</div>
<div id="ref-WDO+15">
<p>Wiśniewski JR., Duś-Szachniewicz K., Ostasiewicz P., Ziółkowski P., Rakus D., Mann M. 2015. Absolute proteome analysis of colorectal mucosa, adenoma, and cancer reveals drastic changes in fatty acid metabolism and plasma membrane transporters. <em>Journal of Proteome Research</em> 14:4005–4018. DOI: <a href="https://doi.org/10.1021/acs.jproteome.5b00523">10.1021/acs.jproteome.5b00523</a>.</p>
</div>
<div id="ref-WKP+14">
<p>Wit M de., Kant H., Piersma SR., Pham TV., Mongera S., Berkel MPA van., Boven E., Pontén F., Meijer GA., Jimenez CR., Fijneman RJA. 2014. Colorectal cancer candidate biomarkers identified by tissue secretome proteome profiling. <em>Journal of Proteomics</em> 99:26–39. DOI: <a href="https://doi.org/10.1016/j.jprot.2014.01.001">10.1016/j.jprot.2014.01.001</a>.</p>
</div>
<div id="ref-XZC+10">
<p>Xie L-Q., Zhao C., Cai S-J., Xu Y., Huang L-Y., Bian J-S., Shen C-P., Lu H-J., Yang P-Y. 2010. Novel proteomic strategy reveal combined <span class="math inline">\(\alpha\)</span>1 antitrypsin and cathepsin D as biomarkers for colorectal cancer early screening. <em>Journal of Proteome Research</em> 9:4701–4709. DOI: <a href="https://doi.org/10.1021/pr100406z">10.1021/pr100406z</a>.</p>
</div>
<div id="ref-YLZ+12">
<p>Yao L., Lao W., Zhang Y., Tang X., Hu X., He C., Hu X., Xu LX. 2012. Identification of EFEMP2 as a serum biomarker for the early detection of colorectal cancer with lectin affinity capture assisted secretome analysis of cultured fresh tissues. <em>Journal of Proteome Research</em> 11:3281–3294. DOI: <a href="https://doi.org/10.1021/pr300020p">10.1021/pr300020p</a>.</p>
</div>
<div id="ref-ZYS+10">
<p>Zhang Y., Ye Y., Shen D., Jiang K., Zhang H., Sun W., Zhang J., Xu F., Cui Z., Wang S. 2010. Identification of transgelin-2 as a biomarker of colorectal cancer by laser capture microdissection and quantitative proteome analysis. <em>Cancer Science</em> 101:523–529. DOI: <a href="https://doi.org/10.1111/j.1349-7006.2009.01424.x">10.1111/j.1349-7006.2009.01424.x</a>.</p>
</div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
