---
title: "Hypoxia"
output:
  html_vignette:
    mathjax: null
vignette: >
  %\VignetteIndexEntry{Hypoxia}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
bibliography: cpdat.bib
csl: peerj.csl
---

```{r setup, include=FALSE}
library(canprot)
library(CHNOSZ)
library(knitr)
## use pngquant to reduce size of PNG images
knit_hooks$set(pngquant = hook_pngquant)
pngquant <- "--speed=1 --quality=0-25"
# in case pngquant isn't available (R-Forge?)
if (!nzchar(Sys.which("pngquant"))) pngquant <- NULL 
```

This vignette shows compositional metrics and phylostrata for proteins that are differentially expressed in hypoxia compared to control conditions.
Abbreviations:

  * <i>Z</i><sub>C</sub> &ndash; carbon oxidation state; <i>n</i><sub>H<sub>2</sub>O</sub> &ndash; stoichiometric hydration state; <i>n</i><sub>O<sub>2</sub></sub> &ndash; stoichiometric oxidation state.
  * <i>n</i><sub>AA</sub> &ndash; protein length; PS &ndash; phylostrata.
  * <i>n</i><sub>down</sub> &ndash; number of down-expressed proteins; <i>n</i><sub>up</sub> &ndash; number of up-expressed proteins.

Stoichiometric values are derived from basis species using the rQEC method (Dick et al., 2020) or from amino acid biosynthetic reactions.
References for gene ages: [Trigos et al. (2017)](https://doi.org/10.1073/pnas.1617743114) (TPPG17); [Liebeskind et al. (2016)](https://doi.org/10.1093/gbe/evw113) (LMM16).

```{r options, echo=FALSE}
options(width = 90)
```

<style type="text/css">
body {
  max-width: 800px;
  margin-left: auto;
  margin-right: auto;
}
</style>

```{r datasets}
datasets <- pdat_hypoxia()
```

```{r comptab, results="hide", message=FALSE, echo = FALSE}
pdat1 <- lapply_canprot(datasets, pdat_hypoxia)
comptab1 <- lapply(pdat1, get_comptab)
pdat2 <- lapply(pdat1, pdat_recomp, "biosynth")
comptab2 <- lapply(pdat2, get_comptab, "nO2")
comptab3 <- lapply(pdat1, get_comptab, "nAA", "PS")
comptab4 <- lapply(pdat1, get_comptab, "nAA", "PS", PS_source = "LMM16")
```

Dashed contour lines in the plots outline the 50% credible region for highest probability density.
Triangle symbols indicate transcriptomic datasets; all others are protein expression.
Abbreviations:

  * Hx48 &ndash; hypoxia 48 h; Hx72 &ndash; hypoxia 72 h.

```{r diffplot, fig.width=8, fig.height=8, fig.align = "center", echo = FALSE, pngquant = pngquant}
par(mfrow = c(2, 2), mar = c(4, 4, 2, 2), mgp = c(2.5, 1, 0))
pch <- ifelse(grepl("transcript", datasets), 2, 1)
diffplot(comptab1, pch = pch)
title(quote("rQEC"~italic(n)[H[2]*O]), font.main = 1)
diffplot(comptab2, c("nO2", "nH2O"), pch = pch)
title(quote("Biosynthetic"~italic(n)[H[2]*O]~"and"~italic(n)[O[2]]), font.main = 1)
diffplot(comptab3, c("nAA", "PS"), pch = pch)
title("Trigos et al. (2017) ages", font.main = 1)
diffplot(comptab4, c("nAA", "PS"), pch = pch)
title("Liebeskind et al. (2016) ages", font.main = 1)
```

In the table, values of &Delta;<i>Z</i><sub>C</sub>, &Delta;<i>n</i><sub>H<sub>2</sub>O</sub> and &Delta;<i>n</i><sub>O<sub>2</sub></sub> are multiplied by 1000, values of &Delta;PS are multiplied by 100, and negative values are shown in bold.

```{r xsummary, results="asis", echo = FALSE}
library(xtable)
out <- xsummary2(comptab1, comptab2, comptab3, comptab4)
# round values and include dataset tags
tags <- sapply(sapply(strsplit(datasets, "="), "[", -1), paste, collapse = ";")
out <- cbind(out[, 1:2], tags = tags, out[, 3:25])
out[, 6:26] <- round(out[, 6:26], 4)
write.csv(out, "hypoxia.csv", row.names = FALSE, quote = 2)
```

## Data Sources
Where given, gene names or other identifiers were converted to UniProt accession numbers using the <a href="https://www.uniprot.org/mapping/">UniProt mapping tool</a>.

__a__. Gene names from Supplemental Tables S1 and S2 of @MRL+05 (genes regulated by both hypoxia and HIF-1).
__b__. Supplemental Table 1 of @SBB+06, filtered to include proteins with expression ratio ≥ 1.7 or ≤ 0.58.
__c__. Gene IDs from Supplementary Table S4 of @BKS+09.
__d__. Supplemental Table 2A of @FWH+13 (control virus cells).
__e__. __f__. Supplemental Table S1 of @RHD+13, filtered to include proteins with iTRAQ ratios < 0.83 or > 1.2 and _p_-value < 0.05.
__g__. Supporting Information table of @VTMF13, filtered to include proteins with a normalized expression ratio of > 1.2 or < 0.83.
__h__. Supporting Information Table S1 @DCH+14.
__i__. __j__. __k__. __l__. Supplemental Table S1 of @DYL+14, filtered to include proteins with _p_-value < 0.05 (-S: supernatant fraction; -P: pellet fraction).
__m__. Gene names from Supporting Information Table S10 of @OHS+14.
__n__. Supplementary Table S1 of @BSA+15.
__o__. Supplemental Information Table S1 of @HWA+16, filtered to include proteins with a fold change of < 0.5 or > 1 and that were detected in only hypoxic or only normoxic conditions.
__p__. __q__. Gene names from @LCS16 (data files provided by Ming-Chih Lai).
__r__. Extracted from Table S2E (whole cell lysate) of @CGH+17, keeping proteins with FDR < 0.05.
__s__. Supplemental Table S4 of @ZXS+17.
__t__. Supplementary Tables S6-S7 (proteome) of @CLY+18.
__u__. List of up- and down-regulated proteins from @GBH+18 (provided by Alex Greenhough), filtered to include proteins with average fold change ≥ 1.5 or ≤ 2/3.
__v__. Gene names from Figure 1F of @LKK+18.
__w__. Supplemental file 1 of @WTG+18 (provided by Gordana Vunjak-Novakovic), filtered to include proteins with Normalized Ratio [Hypoxia MSC/Control MSC] ≥ 1.5 or ≤ 2/3 and _p_-value < 0.05.
__x__. Gene names from Supplementary file S1 of @CSK+19 for the two replicates labelled as "input_Log2ratioHL_firstIP" and "input_Log2ratioLH_secondIP" (soluble extracts before immunoprecipitation), filtered to include proteins where Log2ratio is > 0.2 or < -0.2 for both replicates.
__y__. Proteins identified as up- or down-regulated > 1 SD in Data File S1 of @KAN+19 (sheet "Proteome").
__z__. Gene names from Supplementary Information Tables S6 and S7 of @BCMS20.

## Acknowledgements
Thanks to Alex Greenhough, Ming-Chih Lai, and Gordana Vunjak-Novakovic for providing data files.

## References
