---
title: "Osmotic 2"
output:
  html_vignette:
    mathjax: null
vignette: >
  %\VignetteIndexEntry{Osmotic 2 (with halophiles)}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
bibliography: cpdat.bib
csl: peerj.csl
---

```{r setup, include=FALSE}
library(canprot)
library(CHNOSZ)
library(knitr)
## use pngquant to reduce size of PNG images
knit_hooks$set(pngquant = hook_pngquant)
pngquant <- "--speed=1 --quality=0-25"
# in case pngquant isn't available (R-Forge?)
if (!nzchar(Sys.which("pngquant"))) pngquant <- NULL 
```

This vignette shows compositional metrics for proteins that are differentially expressed in hyperosmotic or hypoosmotic compared to control conditions.
Abbreviations:

  * <i>Z</i><sub>C</sub> &ndash; carbon oxidation state; <i>n</i><sub>H<sub>2</sub>O</sub> &ndash; stoichiometric hydration state; <i>n</i><sub>O<sub>2</sub></sub> &ndash; stoichiometric oxidation state.
  * <i>n</i><sub>AA</sub> &ndash; protein length
  * <i>n</i><sub>down</sub> &ndash; number of down-regulated proteins; <i>n</i><sub>up</sub> &ndash; number of up-regulated proteins.

Stoichiometric values are calculated using basis species (rQEC derivation; [Dick et al., 2020](https://doi.org/10.1101/2020.04.01.020008)) or amino acid biosynthetic reactions.

```{r options, echo=FALSE}
options(width = 90)
```

<style type="text/css">
body {
  max-width: 800px;
  margin-left: auto;
  margin-right: auto;
}
</style>

```{r datasets}
datasets <- pdat_osmotic2(2020)
```

```{r comptab, results="hide", message=FALSE, echo = FALSE}
pdat1 <- lapply_canprot(datasets, pdat_osmotic2)
comptab1 <- lapply(pdat1, get_comptab)
pdat2 <- lapply(pdat1, pdat_recomp, "biosynth")
comptab2 <- lapply(pdat2, get_comptab, "nO2")
comptab3 <- lapply(pdat1, get_comptab, "nAA", "PS")
comptab4 <- lapply(pdat1, get_comptab, "nAA", "PS", PS_source = "LMM16")
```

Dashed contour lines in the plots outline the 50% credible region for highest probability density.
Triangle symbols indicate transcriptomic datasets; all others are protein expression.

```{r diffplot, fig.width=8, fig.height=4, fig.align = "center", echo = FALSE, pngquant = pngquant}
par(mfrow = c(1, 2), mar = c(4, 4, 2, 2), mgp = c(2.5, 1, 0))
pch <- ifelse(grepl("transcript", datasets), 2, 1)
diffplot(comptab1, pch = pch)
title(quote("rQEC"~italic(n)[H[2]*O]), font.main = 1)
diffplot(comptab2, c("nO2", "nH2O"), pch = pch)
title(quote("Biosynthetic"~italic(n)[H[2]*O]~"and"~italic(n)[O[2]]), font.main = 1)
```

In the table, values of &Delta;<i>Z</i><sub>C</sub>, &Delta;<i>n</i><sub>H<sub>2</sub>O</sub> and &Delta;<i>n</i><sub>O<sub>2</sub></sub> are multiplied by 1000, and negative values are shown in bold.

```{r xsummary, results="asis", echo = FALSE}
library(xtable)
out <- xsummary2(comptab1, comptab2, comptab3, comptab4)
# round values and include dataset tags
tags <- sapply(sapply(strsplit(datasets, "="), "[", -1), paste, collapse = ";")
out <- cbind(out[, 1:2], tags = tags, out[, 3:25])
out[, 6:26] <- round(out[, 6:26], 4)
write.csv(out, "osmotic2.csv", row.names = FALSE, quote = 2)
```

## Data Sources

__a__. __b__. Tables 1 and 2 of @LRB+09.
__c__. Supplementary Table 8 of @FTR+10. Only proteins with consistent expression ratios (all > 1 or all < 1) at each time point (15, 60, and 180 min.) were included.
__d__. __e__. Supplementary Tables 2 (membrane proteomics) and 3 (transcriptomics) of @HMO+10. The cytosol proteomics experiment was not included because of the low number of proteins identified with a significant expression difference. Only proteins with consistent expression ratios in at least 3 of 4 time points (10, 30, 60, and 120 min.) were included.
__f__. __g__. Table S-1 of @ZLZ+16. Values of reporter intensities at each condition (6%, 10%, and 17.5% NaCl) were quantile normalized (R function preprocessCore::normalize.quantiles); normalized values were used to compute intensity ratios (10% / 6% NaCl and 17.5% / 10% NaCl). Only proteins with expression ratios > 1.3 in either direction [@ZLZ+16], p-values < 0.05, and at least 2 peptides were included.
__h__. __i__. Tables S2 and S3 of @LLYL17.
__j__. __k__. Tables S1–S6 of @LJC+18. For each of the wild-type and ∆_sigB_ mutant, only proteins that were identified in multicellular vesicles in a single condition (0.5 M salt stress or without salt stress) were included.
__l__. __m__. Supporting Table 1C of @JSP+19. Only proteins with at least 2-fold expression difference and marked as significant were included.

## References
