---
title: "Pancreatic Cancer"
output:
  html_vignette:
    mathjax: null
vignette: >
  %\VignetteIndexEntry{Pancreatic Cancer}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
bibliography: cpdat.bib
csl: peerj.csl
---

```{r setup, include=FALSE}
library(canprot)
library(CHNOSZ)
library(knitr)
## use pngquant to reduce size of PNG images
knit_hooks$set(pngquant = hook_pngquant)
pngquant <- "--speed=1 --quality=0-25"
# in case pngquant isn't available (R-Forge?)
if (!nzchar(Sys.which("pngquant"))) pngquant <- NULL 
```

This vignette shows compositional metrics and phylostrata for proteins that are differentially expressed in pancreatic cancer compared to normal tissue.
Abbreviations:

  * <i>Z</i><sub>C</sub> &ndash; carbon oxidation state; <i>n</i><sub>H<sub>2</sub>O</sub> &ndash; stoichiometric hydration state; <i>n</i><sub>O<sub>2</sub></sub> &ndash; stoichiometric oxidation state.
  * <i>n</i><sub>AA</sub> &ndash; protein length; PS &ndash; phylostrata.
  * <i>n</i><sub>down</sub> &ndash; number of down-regulated proteins; <i>n</i><sub>up</sub> &ndash; number of up-regulated proteins.

Stoichiometric values are derived from basis species using the rQEC method (Dick et al., 2020) or from amino acid biosynthetic reactions.
References for gene ages: [Trigos et al. (2017)](https://doi.org/10.1073/pnas.1617743114) (TPPG17); [Liebeskind et al. (2016)](https://doi.org/10.1093/gbe/evw113) (LMM16).

```{r options, echo=FALSE}
options(width = 90)
```

<style type="text/css">
body {
  max-width: 800px;
  margin-left: auto;
  margin-right: auto;
}
</style>

```{r datasets}
datasets <- pdat_pancreatic(2020)
```

```{r comptab, results="hide", message=FALSE, echo = FALSE}
pdat1 <- lapply_canprot(datasets, pdat_pancreatic)
comptab1 <- lapply(pdat1, get_comptab)
pdat2 <- lapply(pdat1, pdat_recomp, "biosynth")
comptab2 <- lapply(pdat2, get_comptab, "nO2")
comptab3 <- lapply(pdat1, get_comptab, "nAA", "PS")
comptab4 <- lapply(pdat1, get_comptab, "nAA", "PS", PS_source = "LMM16")
```

Dashed contour lines in the plots outline the 50% credible region for highest probability density.
Triangle symbols indicate transcriptomic datasets; all others are protein expression.

```{r diffplot, fig.width=8, fig.height=8, fig.align = "center", echo = FALSE, pngquant = pngquant}
par(mfrow = c(2, 2), mar = c(4, 4, 2, 2), mgp = c(2.5, 1, 0))
pch <- ifelse(grepl("transcript", datasets), 2, 1)
diffplot(comptab1, pch = pch)
title(quote("rQEC"~italic(n)[H[2]*O]), font.main = 1)
diffplot(comptab2, c("nO2", "nH2O"), pch = pch)
title(quote("Biosynthetic"~italic(n)[H[2]*O]~"and"~italic(n)[O[2]]), font.main = 1)
diffplot(comptab3, c("nAA", "PS"), pch = pch)
title("Trigos et al. (2017) ages", font.main = 1)
diffplot(comptab4, c("nAA", "PS"), pch = pch)
title("Liebeskind et al. (2016) ages", font.main = 1)
```

In the table, values of &Delta;<i>Z</i><sub>C</sub>, &Delta;<i>n</i><sub>H<sub>2</sub>O</sub> and &Delta;<i>n</i><sub>O<sub>2</sub></sub> are multiplied by 1000, values of &Delta;PS are multiplied by 100, and negative values are shown in bold.

```{r xsummary, results="asis", echo = FALSE}
library(xtable)
out <- xsummary2(comptab1, comptab2, comptab3, comptab4)
# round values and include dataset tags
tags <- sapply(sapply(strsplit(datasets, "="), "[", -1), paste, collapse = ";")
out <- cbind(out[, 1:2], tags = tags, out[, 3:11])
out[, 6:12] <- round(out[, 6:12], 4)
write.csv(out, "pancreatic.csv", row.names = FALSE, quote = 2)
```

## Data Sources
Gene names or other identifiers were converted to UniProt accession numbers using the <a href="https://www.uniprot.org/mapping/">UniProt mapping tool</a>, except for IPI accession numbers, which were converted using the <a href="https://david-d.ncifcrf.gov/conversion.jsp">DAVID 6.7 conversion tool</a>.

<b>a</b>. Tables 2 and 3 of @LHE+04.
<b>b</b>. Tables 1 and 2 of @CYD+05.
<b>c</b>. Table 2 of @CGB+05.
<b>d</b>. Supplementary Table 1 of @CLC+06.
<b>e</b>. Table 1 of @CTZ+09.
<b>f</b>. IPI numbers from Supplementary Table S2 of @MLC+11.
<b>g</b>. Supplementary Table 3 of @PCS+11.
<b>h</b>. Extracted from the SI Table of @TMW+11.
<b>i</b>. Supplementary Tables 2 and 3 of @KBK+12.
<b>j</b>. SI Table S3 of @KHO+13, filtered to include proteins with an expression ratio >2 [or <0.5] in at least 5 of the 7 experiments and ratio >1 [or <1] in all experiments.
<b>k</b>. Supplementary Table 2 of @KPC+13.
<b>l</b>. <b>m</b>. Supplementary Tables S3 and S4 of @WLL+13a, including proteins with >3/2 or <2/3 fold change in at least 3 of 4 iTRAQ experiments for different pooled samples.
<b>n</b>. Supplementary Tables 2 and 3 of @YKK+13 (data file provided by Youngsoo Kim).
<b>o</b>. SI Table S5 of @ZNWL13.
<b>p</b>. SI Table S5 of @ISI+14, filtered to exclude proteins marked as "not passed", i.e. having inconsistent regulation.
<b>q</b>. Table S6, Sheet 2 of @BZQ+14.
<b>r</b>. Table 1 of @MZH+14.
<b>s</b>. Table S6 of @BHB+15.
<b>t</b>. Supplementary Table of @KKC+16.
<b>u</b>. Supplementary Table S3 of @CHO+18.
<b>v</b>. Table S1 of @SWW+18, filtered to exclude proteins with opposite expression changes in different patients.
<b>w</b>. Gene names extracted from Figure 1b of @ZAH+19.

## Acknowledgement
Thanks to Youngsoo Kim for providing a data file.

## References
