---
title: "Secreted in Hypoxia"
output:
  html_vignette:
    mathjax: null
vignette: >
  %\VignetteIndexEntry{Secreted in Hypoxia}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
bibliography: cpdat.bib
csl: peerj.csl
---

```{r setup, include=FALSE}
library(canprot)
library(CHNOSZ)
library(knitr)
## use pngquant to reduce size of PNG images
knit_hooks$set(pngquant = hook_pngquant)
pngquant <- "--speed=1 --quality=0-25"
# in case pngquant isn't available (R-Forge?)
if (!nzchar(Sys.which("pngquant"))) pngquant <- NULL 
```

This vignette shows compositional metrics and gene ages (phylostrata) for secreted proteins that are differentially expressed in hypoxia compared to control conditions.
Abbreviations:

  * <i>Z</i><sub>C</sub> &ndash; carbon oxidation state; <i>n</i><sub>H<sub>2</sub>O</sub> &ndash; stoichiometric hydration state; <i>n</i><sub>O<sub>2</sub></sub> &ndash; stoichiometric oxidation state.
  * <i>n</i><sub>AA</sub> &ndash; protein length; PS &ndash; phylostrata.
  * <i>n</i><sub>down</sub> &ndash; number of down-regulated proteins; <i>n</i><sub>up</sub> &ndash; number of up-regulated proteins.

Stoichiometric values are derived from basis species using the rQEC method (Dick et al., 2020) or from amino acid biosynthetic reactions.
References for gene ages: [Trigos et al. (2017)](https://doi.org/10.1073/pnas.1617743114) (TPPG17); [Liebeskind et al. (2016)](https://doi.org/10.1093/gbe/evw113) (LMM16).

```{r options, echo=FALSE}
options(width = 90)
```

<style type="text/css">
body {
  max-width: 800px;
  margin-left: auto;
  margin-right: auto;
}
</style>

```{r datasets}
datasets <- pdat_secreted()
```

```{r comptab, results="hide", message=FALSE, echo = FALSE}
pdat1 <- lapply_canprot(datasets, pdat_secreted)
comptab1 <- lapply(pdat1, get_comptab)
pdat2 <- lapply(pdat1, pdat_recomp, "biosynth")
comptab2 <- lapply(pdat2, get_comptab, "nO2")
comptab3 <- lapply(pdat1, get_comptab, "nAA", "PS")
comptab4 <- lapply(pdat1, get_comptab, "nAA", "PS", PS_source = "LMM16")
```

Dashed contour lines in the plots outline the 50% credible region for highest probability density.
Triangle symbols indicate transcriptomic datasets; all others are protein expression.

```{r diffplot, fig.width=8, fig.height=8, fig.align = "center", echo = FALSE, pngquant = pngquant}
par(mfrow = c(2, 2), mar = c(4, 4, 2, 2), mgp = c(2.5, 1, 0))
pch <- ifelse(grepl("transcript", datasets), 2, 1)
diffplot(comptab1, pch = pch)
title(quote("rQEC"~italic(n)[H[2]*O]), font.main = 1)
diffplot(comptab2, c("nO2", "nH2O"), pch = pch)
title(quote("Biosynthetic"~italic(n)[H[2]*O]~"and"~italic(n)[O[2]]), font.main = 1)
diffplot(comptab3, c("nAA", "PS"), pch = pch)
title("Trigos et al. (2017) ages", font.main = 1)
diffplot(comptab4, c("nAA", "PS"), pch = pch)
title("Liebeskind et al. (2016) ages", font.main = 1)
```

In the table, values of &Delta;<i>Z</i><sub>C</sub>, &Delta;<i>n</i><sub>H<sub>2</sub>O</sub> and &Delta;<i>n</i><sub>O<sub>2</sub></sub> are multiplied by 1000, values of &Delta;PS are multiplied by 100, and negative values are shown in bold.

```{r xsummary, results="asis", echo = FALSE}
library(xtable)
out <- xsummary2(comptab1, comptab2, comptab3, comptab4)
# round values and include dataset tags
tags <- sapply(sapply(strsplit(datasets, "="), "[", -1), paste, collapse = ";")
out <- cbind(out[, 1:2], tags = tags, out[, 3:11])
out[, 6:12] <- round(out[, 6:12], 4)
write.csv(out, "secreted.csv", row.names = FALSE, quote = 2)
```

## Data Sources
Gene names or other identifiers were converted to UniProt accession numbers using the <a href="https://www.uniprot.org/mapping/">UniProt mapping tool</a>.

__a__. Tables 2 and 3 of @BRA+10.
__b__. __c__. Gene names from Supplementary Table S1 of @PTD+10, filtered with p-value < 0.05, expression ratio > 1.3 or < 1/1.3 and EF < 2.
__d__. Extracted from Supplementary Table SIII of @JVC+12: median values of peptide quantification (omitting proteins identified with less than 5 peptides that have different signs of log~2~ values); differentially expressed proteins identifed using a log~2~ cutoff of 0.2.
__e__. Table S1 of @KCW+13.
__f__. Extracted from Table 1 of @SKA+13, to include proteins exclusively identified in 1% or 8% O~2~.
__g__. __h__. Extracted Table 1 of @SRS+13a, including unique proteins for 1%, 3%, and 8% O~2~.
__i__. GI numbers from Supplementary Data 6 of @LRS+14.
__j__. __k__. Extracted from Table S1 of @YKK+14, to include proteins identified by at least 2 unique peptides and surpassing a log~2~ cutoff of 0.5 in soluble or exosome fractions.
__l__. __m__. Gene names from Supplementary Information Table 1 of @CRS+15, filtered to include proteins with log~2~ fold change between air and hypoxia > 0.2 or < -0.2.
__n__. Gene names from Tables 1 and 2 of @RSE+16.
__o__. __p__. Extracted from Tables S2A (exosomes) and S2B (secretome) of @CGH+17, keeping proteins with FDR < 0.05.
__q__. Supplementary Tables S8-S9 (secretome) of @CLY+18.
__r__. Supplementary Table 1 of @DWW+18.
__s__. Table S2 of @FPR+18.
__t__. Proteins identified as up- or down-regulated > 1 SD in Data File S1 of @KAN+19 (pooled data from sheets "Soluble Secretome" and "EVs").
__u__. __v__. Extracted from proteinGroups.txt in ProteomeXchange Dataset [PXD008104](ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2019/03/PXD008104) [@NJVS19]. Expression ratios between hypoxia and normoxia were calculated from LFQ intensity values, and proteins were classified as up- or down-expressed if they had expression ratios > 1.2 or < 1/1.2 in all three experiments for one cell type (CAM or NTM). The Majority protein IDs and mean values of the expression ratios were saved in the data file.
__2__. Supplementary Table 2 of @PDT+19, filtered with log~2~ fold-change cutoff of 1.

## References
